perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/widgets/items/ListBox.java

```diff
@@ -0,0 +1,101 @@
+package cz.metacentrum.perun.wui.registrar.widgets.items;
+
+import com.google.gwt.event.dom.client.BlurEvent;
+import com.google.gwt.event.dom.client.BlurHandler;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.ClickHandler;
+import com.google.gwt.user.client.ui.HasVerticalAlignment;
+import com.google.gwt.user.client.ui.HorizontalPanel;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.gwt.user.client.ui.VerticalPanel;
+import cz.metacentrum.perun.wui.widgets.boxes.ExtendedTextBox;
+import cz.metacentrum.perun.wui.widgets.PerunButton;
+import cz.metacentrum.perun.wui.model.beans.ApplicationFormItemData;
+import cz.metacentrum.perun.wui.registrar.widgets.PerunForm;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents ListBox form item.
+ *
+ * @author Jakub Hejda <Jakub.Hejda@cesnet.cz>
+ */
+public class ListBox extends WidgetBox {
+
+	List<ExtendedTextBox> inputList;
+
+	public ListBox(PerunForm form, ApplicationFormItemData item, String lang) {
+		super(form, item, lang);
+	}
+
+	@Override
+	public String getValue() {
+		String value = "";
+		for (ExtendedTextBox input : inputList) {
+			value += (input.getValue() == null || input.getValue().isEmpty()) ? "" : input.getValue() + ",";
```

Comment:
I think we should handle escaping of `,` in the user input here since when it is sent to the backend, it stores raw value and later parsing expects values to be escaped, otherwise it will be considered as more items in the list.

Maybe some alternate logic/method can be implemented for getting list values since you can then use it more safely in validators (without necessity to handle escaping) or as I used it in my local test - just return list of inputs to the validator and validate each value separately.

You will probably need to implement own `ListBoxValidator` to properly handle such validation and probably ditch default validation in `WidgetBox`. Because even if items do not support multiple error messages, we somehow want to validate each input and let user know, which input is wrong from the list, so it needs to reflected in error message.

Also we need that specific SSH keys validation from `SshKeysTextFieldValidator` to be available in this widget by default if destination attribute is `urn:perun:user:attribute-def:def:sshPublicKey` so you want to take it implementation and modify it to expected results in `ListBoxValidator`. Like now it shouldn't be necessary to handle newlines and commas, and error messages should reflect that, but we still need to check the format.